app/src/main/java/com/example/first/hw_adapter_03_03_20/MainActivity.java

```diff
@@ -39,32 +44,39 @@ protected void onCreate(Bundle savedInstanceState) {
 
         nameEt = findViewById(R.id.name_et);
         phoneET = findViewById(R.id.phone_et);
-        type_et = findViewById(R.id.type_et);
         addPerson = findViewById(R.id.add_person);
+        typeRg = findViewById(R.id.type_rg);
 
         addPerson.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 Person person;
-                switch (type_et.getText().toString().toLowerCase()){
-                    case  "s":
+                switch (typeRg.getCheckedRadioButtonId()){
+                    case  R.id.student_rb:
                         person = new Student("Math", 59.0);
+                        addPerson(person , adapter);
```

Comment:
@Netanel6  this is an interesting way of adding the person, why not keep the adapter's function?